static/app/views/issueList/searchBar.spec.tsx

```diff
@@ -0,0 +1,303 @@
+            limit: 3,
+            type: 0,
+          },
+        })
+      );
+    });
+
+    it('cycles through keyboard navigation for selection', async function () {
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar {...defaultProps} />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      const textarea = screen.getByRole('textbox');
+
+      // Keyboard navigate to first item and select
+      userEvent.click(textarea);
+      userEvent.keyboard('t');
+      await waitFor(() =>
+        expect(screen.getAllByTestId('search-autocomplete-item')[0]).toBeInTheDocument()
+      );
+      userEvent.keyboard('{ArrowDown}{Tab}');
+      expect(textarea).not.toHaveValue('t');
+      const firstItemValue = textarea.textContent;
+
+      // Keyboard navigate to second item and select
+      userEvent.keyboard('{selectall}{backspace}');
+      userEvent.keyboard('t');
+      await waitFor(() =>
+        expect(screen.getAllByTestId('search-autocomplete-item')[0]).toBeInTheDocument()
+      );
+      userEvent.keyboard('{ArrowDown}{ArrowDown}{Tab}');
+      expect(textarea).not.toHaveValue(firstItemValue);
+
+      // Keyboard navigate to second item, then back to first item and select
+      userEvent.keyboard('{selectall}{backspace}');
+      userEvent.keyboard('t');
+      await waitFor(() =>
+        expect(screen.getAllByTestId('search-autocomplete-item')[0]).toBeInTheDocument()
+      );
+      userEvent.keyboard('{ArrowDown}{ArrowDown}{ArrowUp}{Tab}');
+      expect(textarea).toHaveValue(firstItemValue);
+    });
+  });
+
+  describe('Pinned Searches', function () {
+    let pinSearch;
+    let unpinSearch;
+
+    beforeEach(function () {
+      MockApiClient.clearMockResponses();
+      pinSearch = MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/pinned-searches/',
+        method: 'PUT',
+        body: {},
+      });
+      unpinSearch = MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/pinned-searches/',
+        method: 'DELETE',
+        body: {},
+      });
+      MockApiClient.addMockResponse({
+        url: '/organizations/org-slug/recent-searches/',
+        method: 'GET',
+        body: [],
+      });
+    });
+
+    it('has pin icon', function () {
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar {...defaultProps} />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      expect(screen.getByTestId('pin-icon')).toBeInTheDocument();
+    });
+
+    it('pins a search from the searchbar', function () {
+      render(
+        <OrganizationContext.Provider value={organization}>
+          <IssueListSearchBar {...defaultProps} query='url:"fu"' />
+        </OrganizationContext.Provider>,
+        {context: routerContext}
+      );
+
+      userEvent.click(screen.getByTestId('pin-icon'));
```

Comment:
Should be able to use `getByRole` here also if it's labeled right

Reply:
Ahh true! https://github.com/getsentry/sentry/pull/40289/commits/12bb1baa669ec644f5b12d41d67bc6981af699b2